ViewModel/Inventory/DefineSIAttributesVM.cs

```diff
@@ -1,492 +1,498 @@
-        {
-            get { return _FirstFloorElevationColumnName; }
-            set { _FirstFloorElevationColumnName = value; NotifyPropertyChanged(); }
-        }
-
 
-
-
-        public string GroundElevationDefaultValue
-        {
-            get { return _GroundElevationDefaultValue; }
-            set { _GroundElevationDefaultValue = value; NotifyPropertyChanged(); }
-        }
-        public bool GroundElevationIsUsingDefault
-        {
-            get { return _GroundElevationIsUsingDefault; }
-            set { _GroundElevationIsUsingDefault = value; NotifyPropertyChanged(); }
-        }
-        public string GroundElevationColumnName
+        private void PathChanged()
         {
-            get { return _GroundElevationColumnName; }
-            set { _GroundElevationColumnName = value; NotifyPropertyChanged(); }
+            UpdateRows();
         }
 
+        private void UpdateRows()
+        {
+            List<string> allColumnNames = GetColumnNames();      
 
+            //required rows
+            _StructureIDRow.Items.Clear();
+            _StructureIDRow.Items.AddRange(allColumnNames);
+            _OccupancyTypeRow.Items.Clear();
+            _OccupancyTypeRow.Items.AddRange(allColumnNames);
+            _FirstFloorElevRow.Items.Clear();
+            _FirstFloorElevRow.Items.AddRange(allColumnNames);
+            _StructureValueRow.Items.Clear();
+            _StructureValueRow.Items.AddRange(allColumnNames);
+            _FoundationHeightRow.Items.Clear();
+            _FoundationHeightRow.Items.AddRange(allColumnNames);
+            _GroundElevRow.Items.Clear();
+            _GroundElevRow.Items.AddRange(allColumnNames);
 
-        public string FoundationHeightDefaultValue
-        {
-            get { return _FoundationHeightDefaultValue; }
-            set { _FoundationHeightDefaultValue = value; NotifyPropertyChanged(); }
-        }
-        public bool FoundationHeightIsUsingDefault
-        {
-            get { return _FoundationHeightIsUsingDefault; }
-            set { _FoundationHeightIsUsingDefault = value; NotifyPropertyChanged(); }
+            //optional rows
+            _ContentValueRow.Items.Clear();
+            _ContentValueRow.Items.AddRange(allColumnNames);
+            _OtherValueRow.Items.Clear();
+            _OtherValueRow.Items.AddRange(allColumnNames);
+            _VehicleValueRow.Items.Clear();
+            _VehicleValueRow.Items.AddRange(allColumnNames);
+            _YearRow.Items.Clear();
+            _YearRow.Items.AddRange(allColumnNames);
+            _ModuleRow.Items.Clear();
+            _ModuleRow.Items.AddRange(allColumnNames);
         }
 
-        public string FoundationHeightColumnName
+        private List<string> GetColumnNames()
         {
-            get { return _FoundationHeightColumnName; }
-            set { _FoundationHeightColumnName = value; NotifyPropertyChanged(); }
+            DbfReader dbf = new DbfReader(System.IO.Path.ChangeExtension(_Path, ".dbf"));
```

Comment:
Unnecessary namespacing

Reply:
I did, but then it was in error because i have a property called "Path". I am keeping the namespace for this line.